### ISSUE TITLE
Implement Printer Error Validation and Paper Roll Status Check

### ISSUE THREAD

**Issue by taylorlacort** [DantSu/ESCPOS-ThermalPrinter-Android#528]
**Add functionalities to check printer errors such as paper roll status and other potential issues using ESC/POS commands.**

**Tasks:**

Add ESC/POS Commands for Status Check:

- Implement ESC/POS commands to query the printer status.
- Interpret responses to identify errors like paper out, head open, etc.
- Create checkPrinterStatus Function:

- Develop a function to send status commands to the printer.
- Analyze the response and return the printer status.
- Update Documentation:

- Add usage examples of the new function to the documentation.
- Provide instructions on interpreting different status responses.
- Test on Various Printer Models:

Test the new functionality on different ESC/POS printer models to ensure compatibility.

**Comment by hugofialho** [comment]
This functionality will be great!

**Comment by dshalaby** [comment]
Yes, this would be a great feature! So that way our software can alert the user of printer issues needing attention in the use case where the printer isn't located directly in front of them, like when something is printed to a kitchen from the front of house.
